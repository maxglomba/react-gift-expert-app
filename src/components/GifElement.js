import React from 'react';
import PropTypes from 'prop-types';

function GifElement({ id, url, title, animation_delay = 0 }) {
  
  return (
    <div className='thumbCard animate__animated animate__fadeIn animate__slow'>
        <img src={url} alt={title} />
        
        <p>{title}</p>
    </div>
  )
}

GifElement.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

export default GifElement
