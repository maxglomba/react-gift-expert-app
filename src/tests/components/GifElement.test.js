import React from 'react';
import { shallow } from 'enzyme'
import GifElement from '../../components/GifElement';


describe('Pruebas en componente <GifElement />', () => { 
    const url = 'https://media4.giphy.com/media/yovOUEWBV2R46yrQ0B/giphy.gif?cid=b89f5553wrq1wehi68egkpgpghy2ab63egohqzk6t49644ym&rid=giphy.gif&ct=g';
    const title = 'Test title';
    const wrapper = shallow(<GifElement url={url} title={title} />);

    test('Debe renderizar el componente correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe colocar el titulo de imagen correctamente', () => { 
        const titleText = wrapper.find('p').text().trim();
        expect(titleText).toBe(title);
    })

    test('Debe colocar el url y alt de imagen correctamente', () => { 
        const {src, alt} = wrapper.find('img').props();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe tener la animacion de fadeIn', () => { 
        const hasClass = wrapper.find('div').hasClass('animate__fadeIn');
        expect(hasClass).toBe(true);
    })

 })