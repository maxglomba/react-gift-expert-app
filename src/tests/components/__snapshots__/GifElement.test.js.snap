// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifElement /> Debe renderizar el componente correctamente 1`] = `
<div
  className="thumbCard animate__animated animate__fadeIn animate__slow"
>
  <img
    alt="Test title"
    src="https://media4.giphy.com/media/yovOUEWBV2R46yrQ0B/giphy.gif?cid=b89f5553wrq1wehi68egkpgpghy2ab63egohqzk6t49644ym&rid=giphy.gif&ct=g"
  />
  <p>
    Test title
  </p>
</div>
`;
